lib/address.py

```diff
@@ -311,24 +308,55 @@ def validate(cls, pubkey):
 
     @cachedproperty
     def address(self):
-        '''Convert to an Address object.'''
+        '''Convert to a P2PKH Address object.'''
         return Address(hash160(self.pubkey), Address.ADDR_P2PKH)
 
     def is_compressed(self):
         '''Returns True if the pubkey is compressed.'''
         return len(self.pubkey) == 33
 
     def to_ui_string(self):
-        '''Convert to a hexadecimal string.'''
+        '''Convert to a hexadecimal string.
+
+        (do not change this -- many places rely on this to be hexadecimal)
+        '''
         return self.pubkey.hex()
 
     def to_storage_string(self):
         '''Convert to a hexadecimal string for storage.'''
         return self.pubkey.hex()
 
+    def to_P2PKH_script(self):
+        '''Return a P2PKH script.'''
+        return self.address.to_script()
+
+    def __str__(self):
+        return self.to_ui_string()
+
+    def __repr__(self):
+        return '<PubKey {}>'.format(self.__str__())
+
+
+class P2PKOutput(namedtuple("P2PKTuple", "public_key")):
+
+    @classmethod
+    def from_pubkey(cls, public_key):
+        assert isinstance(public_key, PublicKey)
+        self.public_key = public_key
```

Comment:
Um. `@classmethod`.. no self...

Reply:
yes sir

Reply:
I meant to say __init__ I promise!

Reply:
wait no that's backwards heh... NamedTuples and their weirdness man

Reply:
I should hope you meant.. `return P2PKOutput(public_key)` .. or somesuch... ;)